src/__test__/e2e/api.test.ts

```diff
@@ -0,0 +1,161 @@
+import Common, { Chain, Hardfork } from '@ethereumjs/common';
+import { TransactionFactory } from '@ethereumjs/tx';
+import { question } from 'readline-sync';
+import { encode } from 'rlp';
+import {
+  fetchActiveWallets,
+  fetchAddresses,
+  pair,
+  signBtcLegacyTx,
+  signBtcSegwitTx,
+  signBtcWrappedSegwitTx,
+  signMessage,
+} from '../../api';
+import { HARDENED_OFFSET } from '../../constants';
+import { BTC_PURPOSE_P2SH_P2WPKH, BTC_TESTNET_COIN } from '../utils/helpers';
+import { dexlabProgram } from './signing/__mocks__/programs';
+import {
+  addAddressTags,
+  fetchAddressTags,
+  fetchLedgerLiveAddresses,
+  removeAddressTags,
+  sign,
+  signSolanaTx,
+} from '../../api/index';
+import { setupClient } from '../utils/setup';
+
+describe('API', () => {
+  test('pair', async () => {
+    const isPaired = await setupClient();
+    if (!isPaired) {
+      const secret = question('Please enter the pairing secret: ');
+      await pair(secret.toUpperCase());
+    }
+  });
+
+  describe('signing', () => {
+    describe('bitcoin', () => {
+      const btcTxData = {
+        prevOuts: [
+          {
+            txHash:
+              '6e78493091f80d89a92ae3152df7fbfbdc44df09cf01a9b76c5113c02eaf2e0f',
+            value: 10000,
+            index: 1,
+            signerPath: [
+              BTC_PURPOSE_P2SH_P2WPKH,
+              BTC_TESTNET_COIN,
+              HARDENED_OFFSET,
+              0,
+              0,
+            ],
+          },
+        ],
+        recipient: 'mhifA1DwiMPHTjSJM8FFSL8ibrzWaBCkVT',
+        value: 1000,
+        fee: 1000,
+        changePath: [
+          BTC_PURPOSE_P2SH_P2WPKH,
+          BTC_TESTNET_COIN,
+          HARDENED_OFFSET,
+          1,
+          0,
+        ],
+      };
+      test('legacy', async () => {
+        await signBtcLegacyTx(btcTxData);
+      });
+
+      test('segwit', async () => {
+        await signBtcSegwitTx(btcTxData);
+      });
+
+      test('wrapped segwit', async () => {
+        await signBtcWrappedSegwitTx(btcTxData);
+      });
+    });
+
+    describe('ethereum', () => {
+      const txData = {
+        type: 1,
+        maxFeePerGas: 1200000000,
+        maxPriorityFeePerGas: 1200000000,
+        nonce: 0,
+        gasLimit: 50000,
+        to: '0xe242e54155b1abc71fc118065270cecaaf8b7768',
+        value: 1000000000000,
+        data: '0x17e914679b7e160613be4f8c2d3203d236286d74eb9192f6d6f71b9118a42bb033ccd8e8',
+        gasPrice: 1200000000,
+      };
+
+      test('message', async () => {
+        await signMessage('test message');
```

Comment:
Can you infer whether the message is `personal_sign` or EIP712 based on the argument itself? I can't remember how EIP712 requests are typed but I belive they can be JSON strings. Also note that personl_sign can be JSON strings. That's why you normally pass the message type

Reply:
I didn't fully understand how messages were being handled. 

I swapped to checking if something is not a string with the specific properties of an EIP712 payload to set the protocol [here](https://github.com/GridPlus/gridplus-sdk/pull/525/commits/235fd9e7f44a6044fb7e5e6f850b64af23c5b4fc#diff-9779491ad4be49b873990e462d086864132c46713b7d7a94798ef36fac412db5R130).